File: 30 Day Challenge/April 2020/Week 1-1.cpp
#include <iostream>
#include <vector>
#include <bits/stdc++.h>
#include <algorithm>

using namespace std;

class Solution {
public:
    int singleNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        for(int i = 0; i < nums.size() - 1; i += 2) {
            if(nums[i] != nums[i+1])
                return nums[i];
        }
        return nums[nums.size()-1];
    }
};

int main() {
    Solution * s = new Solution();
    vector<int> input;
    // Size of input array
    int n;
    cin >> n;
    // Input each element of an array
    while(n--) {
        int t;
        cin >> t;
        input.push_back(t);
    }
    cout << s->singleNumber(input) << endl;
}

File: 30 Day Challenge/April 2020/Week 1-1.py
from typing import List

class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        i = 0
        nums.sort()
        while i < len(nums)-1:
            if nums[i] != nums[i+1]:
                return nums[i]
            i += 2
        return nums[-1]

if __name__ == "__main__":
    s = Solution()
    # Get input for number of elements in array
    n = int(input())
    arr = []
    # Get each element of array as input
    for i in range(n):
        arr.append(int(input()))
    print(s.singleNumber(arr))